### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Solobarine**
This pull request is to review the views added to the project.
The styling was done to match the wireframe design.
Some features implemented include:
- Partials
- Layout
- Styling

### CODE REVIEW

**Review by rotimiazeez — COMMENTED** [review]
# Hi @Solobarine , 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that is failing some linter checks, please fix this and submit again 😉

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: app/views/users/_user.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="short_user">
+    <div class="profile_image"></div>
+    <div class="short_bio">
+        <h3>Username</h3>
+        <p>Number of Posts  <small>x</small></p>
+    </div>
+</div>
```
Comment (franclobo) [inline app/views/users/_user.html.erb:7]: - [ ] Well done using partials, but all the information in the blog should come from rails console or `./db/seed.rb`, creating users, posts, and comments in this way will allow adding the correct information of each class, please 🙏🏽 refactor your code to be able to pol¿pulate the blog with the info from rails console or seeds. This partial is not necessary but `index.html.erb` and `show.html.erb`

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,8 @@
-<h1>This a list of all users</h1>
\ No newline at end of file
+<section id="all_users">
+    <h1>This a list of all users</h1>
+    <div>
+        <% 5.times do %>
+            <%= render 'user' %>
+        <% end %>
```
Comment (franclobo) [inline app/views/users/index.html.erb]: - [ ] Please 🙏🏽 refactor this code to be able to populate the blog with the info from rails console or seeds, it allows to create new users with the correct information.
- [ ] The user name should be a link to the user show page.

File: app/views/users/user.html.erb [hunk 3]
```diff
@@ -1 +1,10 @@
-<h1>This is the details of a User</h1>
\ No newline at end of file
+<section id="single_user">
+    <%= render 'user' %>
+    <p id="bio">Lorem ipsum dolor sit, amet consectetur adipisicing elit. Voluptatum excepturi corporis doloremque eum sint itaque ex dignissimos dolorem quasi quam vitae nobis eligendi, a explicabo, non eaque ducimus illum. Necessitatibus ipsum libero, rerum praesentium ad, aliquid rem suscipit doloremque quam itaque sed nemo illum officia. Explicabo pariatur nostrum quas voluptatibus.</p>
+    <div id="user_posts">
+        <% 3.times do %>
+            <%= render 'posts/post'%>
+        <% end %>
```
Comment (franclobo) [inline app/views/users/user.html.erb]: - [ ] Please 🙏🏽 refactor this code to be able to populate the blog with the info from rails console or seeds, it allows the creation of users' correct info.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,12 @@
...
+    <h1>This is all the users posts</h1>
+    <%= render 'users/user' %>
+    <% 2.times do%>
+        <%= render 'post' %>
+        <div class="post_item">
+            <% 5.times do%>
+                <%= render 'comment' %>
+            <% end %>
+        </div>
+    <% end %>
```
Comment (franclobo) [inline app/views/posts/index.html.erb]: - [ ] Please 🙏🏽 refactor this code to be able to populate the blog with the info from rails console or seeds, it allows the creation of posts with correct info.
- [ ] The post title should be a link that returns the post information, its comments, and likes.

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,14 @@
...
+    </div>
+    <div id="post_comments">
+        <% 10.times do %>
+            <%= render 'comment' %>
+        <% end %>
```
Comment (franclobo) [inline app/views/posts/show.html.erb]: - [ ] Please 🙏🏽 refactor this code to be able to populate the blog with the info from rails console or seeds, it allows the creation of posts' show page with correct info.

File: app/views/posts/_post.html.erb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<div class="post_preview">
+    <h4>Post #</h4>
+    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Aspernatur animi id, dolore magni ipsa, magnam beatae dolor aliquam minus temporibus eius, sit quos nisi incidunt autem nobis iusto. Explicabo, eaque!</p>
+    <small>Comments x, likes y</small>
+</div>
```
Comment (franclobo) [inline app/views/posts/_post.html.erb]: - [ ] As I said before it is not necessary partials but the `index.html.erb` and `show.html.erb` files.

File: app/views/posts/_comment.html.erb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<div class="comment">
+    <p>Username: Comment #</p>
+</div>
```
Comment (franclobo) [inline app/views/posts/_comment.html.erb:3]: - [ ] As I said before it is not necessary partials but the `index.html.erb` and `show.html.erb` files.

File: app/views/users/user.html.erb [hunk 8]
```diff
@@ -1 +1,10 @@
...
+    </div>
+   <button>See All Posts</button>
```
Comment (franclobo) [inline app/views/users/user.html.erb]: - [ ] Please 🙏🏽 refactor this code, the button should link to the posts page to see all posts.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Solobarine, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] There is an error in `db/migrate/20230503003631_add_post_to_comment.rb` that interrupt the db migrations, please 🙏🏽 fix this issue to allow run the migrations 👇🏽 

image

- Please 🙏🏽 refactor the root to see the list of user at the beginning instead of the Ruby on Rails logo for a better UX. I recommend using `root "users#index"` in `./config/routes.rb`

- Please 🙏🏽 after refactoring your code ensure that all pages open successfully, the posts#show page returned this issue at the moment 👇🏽 

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Solobarine, 
congratulations-1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Optional suggestions
- [ ] Optionally kindly think about centering you design so as to match the given design
- [ ] Also think about getting rid of this data that is displaying above the `see all posts` button
image



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**